typescript-carousel/main.ts

```diff
@@ -0,0 +1,116 @@
+interface PokemonImg {
+  URL: string;
+  alt: string;
+  class: string;
+  id: string;
+}
+
+const $carousel = document.querySelector(
+  '.carousel-container'
+) as HTMLDivElement;
+const $imageNodeList = document.querySelectorAll(
+  '.carousel-img'
+) as NodeListOf<HTMLImageElement>;
+const $circleNodeList = document.querySelectorAll(
+  '.fa-circle'
+) as NodeListOf<HTMLElement>;
+
+if (!$carousel) throw new Error('no carousel found');
+if (!$imageNodeList) throw new Error('no image node list found');
+if (!$circleNodeList) throw new Error('no circle node list found');
+
+const images: PokemonImg[] = [
+  {
+    URL: 'images/001.png',
+    alt: 'Bulbasaur',
+    class: 'carousel-img',
+    id: 'bulbasaur',
```

Comment:
What is this array doing for you? It looks like you're only using it for its length to determine if you should set your index to 5 or 0 if you go forward or back at the end. You've queried for all of your `$imageNodeList` elements so why not use that to determine your length?